salt/vector/install.sls

```diff
@@ -13,3 +13,9 @@ ensure_state_of_systemd_service_file:
   file.managed:
     - name: /etc/systemd/system/vector.service
     - source: salt://vector/files/vector.service
+  {% if salt.grains.get('init') == 'systemd' %}
```

Comment:
There's not really a need for this check. We're already assuming systemd by the contents of the unit file.

Reply:
OK. I'd copied the logic from the cassandra state.